kwave/kWaveSimulation.py

```diff
@@ -503,7 +521,7 @@ def input_checking(self, calling_func_name) -> None:
         self.check_calling_func_name_and_dim(calling_func_name, k_dim)
 
         # run subscript to check optional inputs
-        self.options = SimulationOptions.option_factory(self.kgrid, self.elastic_code, self.axisymmetric, **self.kwargs)
+        self.options = SimulationOptions.option_factory(self.kgrid, self.elastic_code, self.axisymmetric, self.options)
```

Comment:
This logic can be improved in the future. Changing it here will introduce too many changes at once.

Reply:
👍